ngraph/frontend/paddlepaddle/src/op/pool2d.cpp

```diff
@@ -25,47 +15,227 @@ namespace ngraph
+
+                        if (stride_h < 1 || stride_w < 1)
+                        { // upsampling?
+                            throw std::runtime_error("Unsupported pooling adaptive type!");
+                        }
+
+                        if (pooling_type == "max")
+                        {
+                            std::cout << "Convert MAx" << std::endl;
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::MaxPool>(
+                                    data,
+                                    ngraph::Strides{stride_h, stride_w},
+                                    pad_begin,
+                                    pad_end,
+                                    ngraph::Shape{kernel_h, kernel_w},
+                                    rounding_type,
+                                    auto_pad)},
+                                {"Out"});
+                        }
+                        else
+                        {
+                            bool exclude_pad = node.get_attribute<bool>("exclusive") ? true : false;
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::AvgPool>(
+                                    data,
+                                    ngraph::Strides{stride_h, stride_w},
+                                    pad_begin,
+                                    pad_end,
+                                    ngraph::Shape{kernel_h, kernel_w},
+                                    exclude_pad,
+                                    rounding_type,
+                                    auto_pad)},
+                                {"Out"});
+                        }
                     }
                     else
                     {
-                        throw std::runtime_error("Unsupported pooling type");
+                        auto strides = node.get_attribute<std::vector<int32_t>>("strides");
+                        auto paddings = node.get_attribute<std::vector<int32_t>>("paddings");
+
+                        uint64_t kernel_h, kernel_w;
+                        if (kernel_shape.size() == 1)
+                        {
+                            kernel_h = kernel_w = kernel_shape[0];
+                        }
+                        else
+                        {
+                            kernel_h = kernel_shape[0];
+                            kernel_w = kernel_shape[1];
+                        }
+
+                        if ((input_h > 0) && (input_h + paddings[0] < kernel_h))
+                        {
+                            kernel_h = input_h + paddings[0];
+                        }
+                        if ((input_w > 0) && (input_w + paddings[1] < kernel_w))
+                        {
+                            kernel_w = input_w + paddings[1];
```

Comment:
condition is not covered by unit test

Reply:
@ceciliapeng2011 please add coverage test